lib/chat/chat_messenger.dart

```diff
@@ -78,11 +73,36 @@ class ChatMessengerState extends State<ChatMessenger> {
     );
   }
 
+  Future uploadImage() async {
+    String fileName = DateTime.now().millisecondsSinceEpoch.toString();
+    StorageReference reference = FirebaseStorage.instance.ref().child(fileName);
+    StorageUploadTask uploadTask = reference.putFile(imageFile);
+    StorageTaskSnapshot storageTaskSnapshot = await uploadTask.onComplete;
+    storageTaskSnapshot.ref.getDownloadURL().then((downloadUrl) {
+      imageUrl = downloadUrl as String;
+      submitMessage(1, imageUrl);
```

Comment:
I noticed you're passing `1` to indicate a type in `submitMessage` but I'm confused as to what the distinction between types are. It might be a good idea to define these types in a constants file and import in this file to be used for readability.